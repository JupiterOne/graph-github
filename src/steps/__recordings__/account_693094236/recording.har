{
  "log": {
    "_recordingName": "account",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "7c34bb6b06a875dd6730e58f9ebd212d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.machine-man-preview+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/18.12.0 octokit-core.js/3.5.1 Node.js/16.14.2 (darwin; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 813,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://api.github.com/app/installations/28077570/access_tokens"
        },
        "response": {
          "bodySize": 791,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 791,
            "text": "{\"token\":\"[REDACTED]\",\"expires_at\":\"2050-12-31T18:09:20Z\",\"permissions\":{\"members\":\"read\",\"organization_administration\":\"write\",\"organization_dependabot_secrets\":\"read\",\"organization_hooks\":\"read\",\"organization_plan\":\"read\",\"organization_projects\":\"read\",\"organization_secrets\":\"read\",\"organization_self_hosted_runners\":\"read\",\"organization_user_blocking\":\"read\",\"team_discussions\":\"read\",\"actions\":\"read\",\"administration\":\"write\",\"contents\":\"read\",\"environments\":\"read\",\"issues\":\"read\",\"metadata\":\"read\",\"pull_requests\":\"read\",\"secrets\":\"read\",\"single_file\":\"read\",\"vulnerability_alerts\":\"read\"},\"repository_selection\":\"all\",\"single_file\":\"CODEOWNER\",\"has_multiple_single_files\":true,\"single_file_paths\":[\"CODEOWNER\",\"package.json\",\"package.lock\",\"yarn.lock\"]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 11 Aug 2022 16:30:13 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "791"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "etag",
              "value": "\"1820f94d29c85bb574173241f063794028699b2df8719efd13a5e06b56d31471\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; param=machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "C541:72EF:19E24CB:2FFD567:62F52E95"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1073,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 201,
          "statusText": "Created"
        },
        "startedDateTime": "2022-08-11T16:30:13.149Z",
        "time": 389,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 389
        }
      },
      {
        "_id": "81e94b10a940fee49cf30302bf46eb4c",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/18.12.0 octokit-core.js/3.5.1 Node.js/16.14.2 (darwin; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 762,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.github.com/app/installations/28077570"
        },
        "response": {
          "bodySize": 1039,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1039,
            "text": "{\"id\":28077570,\"account\":{\"login\":\"j1-ingest\",\"id\":100235621,\"node_id\":\"O_kgDOBfl5ZQ\",\"avatar_url\":\"https://avatars.githubusercontent.com/u/100235621?v=4\",\"gravatar_id\":\"\",\"url\":\"https://api.github.com/users/j1-ingest\",\"html_url\":\"https://github.com/j1-ingest\",\"followers_url\":\"https://api.github.com/users/j1-ingest/followers\",\"following_url\":\"https://api.github.com/users/j1-ingest/following{/other_user}\",\"gists_url\":\"https://api.github.com/users/j1-ingest/gists{/gist_id}\",\"starred_url\":\"https://api.github.com/users/j1-ingest/starred{/owner}{/repo}\",\"subscriptions_url\":\"https://api.github.com/users/j1-ingest/subscriptions\",\"organizations_url\":\"https://api.github.com/users/j1-ingest/orgs\",\"repos_url\":\"https://api.github.com/users/j1-ingest/repos\",\"events_url\":\"https://api.github.com/users/j1-ingest/events{/privacy}\",\"received_events_url\":\"https://api.github.com/users/j1-ingest/received_events\",\"type\":\"Organization\",\"site_admin\":false},\"repository_selection\":\"all\",\"access_tokens_url\":\"https://api.github.com/app/installations/28077570/access_tokens\",\"repositories_url\":\"https://api.github.com/installation/repositories\",\"html_url\":\"https://github.com/organizations/j1-ingest/settings/installations/28077570\",\"app_id\":226546,\"app_slug\":\"adaminict-test\",\"target_id\":100235621,\"target_type\":\"Organization\",\"permissions\":{\"issues\":\"read\",\"actions\":\"read\",\"members\":\"read\",\"secrets\":\"read\",\"contents\":\"read\",\"metadata\":\"read\",\"single_file\":\"read\",\"environments\":\"read\",\"pull_requests\":\"read\",\"administration\":\"write\",\"team_discussions\":\"read\",\"organization_plan\":\"read\",\"organization_hooks\":\"read\",\"organization_secrets\":\"read\",\"vulnerability_alerts\":\"read\",\"organization_projects\":\"read\",\"organization_user_blocking\":\"read\",\"organization_administration\":\"write\",\"organization_dependabot_secrets\":\"read\",\"organization_self_hosted_runners\":\"read\"},\"events\":[],\"created_at\":\"2022-08-08T18:43:57.000Z\",\"updated_at\":\"2022-08-08T18:52:45.000Z\",\"single_file_name\":\"CODEOWNER\",\"has_multiple_single_files\":true,\"single_file_paths\":[\"CODEOWNER\",\"package.json\",\"package.lock\",\"yarn.lock\"],\"suspended_by\":null,\"suspended_at\":null}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 11 Aug 2022 16:30:14 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "etag",
              "value": "W/\"9bcda4677952608a11789e40311d28e8bef071b90a48b5c26423c50e3e68b2ad\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "C543:27DC:18A37B9:2DF76D6:62F52E96"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1079,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-11T16:30:13.568Z",
        "time": 505,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 505
        }
      },
      {
        "_id": "22e4c882794d943e8a6f0961702d5000",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 443,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiterone-graph-github"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "443"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 328,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json; charset=utf-8",
            "params": [],
            "text": "{\"query\":\"\\n    query ($login: String!) {\\n      organization(login: $login) {\\n        id\\n        login\\n        name\\n        createdAt\\n        updatedAt\\n        description\\n        email\\n        databaseId\\n        isVerified\\n        location\\n        websiteUrl\\n        url\\n      }\\n      ...on Query {\\n    rateLimit {\\n      limit\\n      cost\\n      remaining\\n      resetAt\\n    }\\n  }\\n    }\",\"variables\":{\"login\":\"j1-ingest\"}}"
          },
          "queryString": [],
          "url": "https://api.github.com/graphql"
        },
        "response": {
          "bodySize": 539,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 539,
            "text": "{\"data\":{\"organization\":{\"id\":\"O_kgDOBfl5ZQ\",\"login\":\"j1-ingest\",\"name\":\"JupiterOne GitHub Ingest Test Organization\",\"createdAt\":\"2022-02-22T19:40:36Z\",\"updatedAt\":\"2022-03-25T14:49:00Z\",\"description\":\"This org houses test data that can be used when developing against the JupiterOne/graph-github repository.\",\"email\":\"support@jupiterone.com\",\"databaseId\":100235621,\"isVerified\":false,\"location\":\"United States of America\",\"websiteUrl\":null,\"url\":\"https://github.com/j1-ingest\"},\"rateLimit\":{\"limit\":5000,\"cost\":1,\"remaining\":4690,\"resetAt\":\"2022-08-11T17:24:43Z\"}}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Thu, 11 Aug 2022 16:30:14 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; format=json"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "4690"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1660238683"
            },
            {
              "name": "x-ratelimit-used",
              "value": "310"
            },
            {
              "name": "x-ratelimit-resource",
              "value": "graphql"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "x-github-request-id",
              "value": "C549:3F8D:1114D42:31BA987:62F52E96"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1086,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-11T16:30:14.086Z",
        "time": 214,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 214
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
