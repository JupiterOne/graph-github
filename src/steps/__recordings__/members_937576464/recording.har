{
  "log": {
    "_recordingName": "members",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "06d6593fc4a07c98683e9f0972b1c5e0",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.machine-man-preview+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/18.12.0 octokit-core.js/3.5.1 Node.js/16.14.2 (darwin; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 813,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://api.github.com/app/installations/23522163/access_tokens"
        },
        "response": {
          "bodySize": 102,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 102,
            "text": "{\"message\":\"A JSON web token could not be decoded\",\"documentation_url\":\"https://docs.github.com/rest\",\"token\":\"[REDACTED]\",\"expires_at\":\"2050-12-31T18:09:20Z\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Fri, 02 Sep 2022 15:59:41 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "102"
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; param=machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "x-github-request-id",
              "value": "F4B9:689B:57D06:D196E:6312286D"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 939,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 401,
          "statusText": "Unauthorized"
        },
        "startedDateTime": "2022-09-02T15:59:40.909Z",
        "time": 160,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 160
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
