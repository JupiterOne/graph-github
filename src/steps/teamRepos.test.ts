jest.setTimeout(20000);

import {
  Recording,
  executeStepWithDependencies,
} from '@jupiterone/integration-sdk-testing';
import { buildStepTestConfig } from '../../test/config';
import { setupGithubRecording } from '../../test/recording';
import { Steps } from '../constants';

let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

test('fetchTeamRepos exec handler', async () => {
  recording = setupGithubRecording({
    directory: __dirname,
    name: 'teamRepos',
  });
  const stepConfig = buildStepTestConfig(Steps.FETCH_TEAM_REPOS);
  const stepResults = await executeStepWithDependencies(stepConfig);

  expect(stepResults).toMatchStepMetadata(stepConfig);
});
