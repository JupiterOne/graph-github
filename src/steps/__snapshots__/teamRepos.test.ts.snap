// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTeamRepos exec handler 1`] = `
Object {
  "collectedEntities": Array [],
  "collectedRelationships": Array [
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG5ypUQ",
      "_key": "R_kgDOG5ypUQ|allows|T_kwDOBfl5Zc4AV-Cl",
      "_toEntityKey": "T_kwDOBfl5Zc4AV-Cl",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG-zMhA",
      "_key": "R_kgDOG-zMhA|allows|T_kwDOBfl5Zc4AV-Cl",
      "_toEntityKey": "T_kwDOBfl5Zc4AV-Cl",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG5ypUQ",
      "_key": "R_kgDOG5ypUQ|allows|T_kwDOBfl5Zc4AWFtb",
      "_toEntityKey": "T_kwDOBfl5Zc4AWFtb",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG_MLeg",
      "_key": "R_kgDOG_MLeg|allows|T_kwDOBfl5Zc4AWFtb",
      "_toEntityKey": "T_kwDOBfl5Zc4AWFtb",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG5ypUQ",
      "_key": "R_kgDOG5ypUQ|allows|T_kwDOBfl5Zc4AWFta",
      "_toEntityKey": "T_kwDOBfl5Zc4AWFta",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG-zMhA",
      "_key": "R_kgDOG-zMhA|allows|T_kwDOBfl5Zc4AWFta",
      "_toEntityKey": "T_kwDOBfl5Zc4AWFta",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": true,
      "role": "WRITE",
      "triagePermission": true,
    },
    Object {
      "_class": "ALLOWS",
      "_fromEntityKey": "R_kgDOG_MJdQ",
      "_key": "R_kgDOG_MJdQ|allows|T_kwDOBfl5Zc4AWFta",
      "_toEntityKey": "T_kwDOBfl5Zc4AWFta",
      "_type": "github_repo_allows_team",
      "adminPermission": false,
      "displayName": "ALLOWS",
      "maintainPermission": false,
      "pullPermission": true,
      "pushPermission": false,
      "role": "READ",
      "triagePermission": false,
    },
  ],
  "encounteredTypes": Array [
    "github_account",
    "github_repo",
    "github_team",
    "github_account_owns_repo",
    "github_account_has_team",
    "github_repo_allows_team",
  ],
  "numCollectedEntities": 0,
  "numCollectedRelationships": 7,
}
`;
